### ISSUE TITLE
inno

### ISSUE THREAD

**Issue by 666tugege** [666tugege/mmdetection#1]
### 1. If I want to train the COCO dataset based on queryinst network, replace its backbone with regnet and freeze the  2 stages of the backbone during training (weights are not updated), what changes need to be made?

- change the backbone(regent) in config file(mmdetection/configs/queryinst/):
```
model= dict(
    type='QueryInst',
    backbone=dict(
        type='regnet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=2,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch'),

```

**Comment by 666tugege** [comment]
hello~

**Comment by 666tugege** [comment]
### 2. Based on answer 1, add at least 3 kinds of data enhancement

- add train_pipeline in config file(mmdetection/configs/queryinst/):
```
model = dict(
    type='QueryInst',
    pretrained='torchvision://regnet',
    backbone=dict(
        train_pipeline = [
        dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
        dict(type='RandomFlip', flip_ratio=0.5),
        dict(type='Normalize', **img_norm_cfg),
        ]
```

**Comment by 666tugege** [comment]
### 3. Based on answer 2, the categories of the dataset are changed to "eca_wl, eca_nbi, eca_iod, gm_up, gm_down, g_ulcer", what changes are needed?

- (1) Add the classes fields in `data.train,` `data.val` and `data.test`.
```
dataset_type = 'MyDataset'
classes = ('eca_wl', 'eca_nbi', 'eca_iod', 'gm_up', 'gm_down', 'g_ulcer')
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        classes=classes,
        ann_file='path/to/your/train/annotation_data',
        img_prefix='path/to/your/train/image_data'),
    val=dict(
        type=dataset_type,
        classes=classes,
        ann_file='path/to/your/val/annotation_data',
        img_prefix='path/to/your/val/image_data'),
    test=dict(
        type=dataset_type,
        classes=classes,
        ann_file='path/to/your/test/annotation_data',
        img_prefix='path/to/your/test/image_data'))
```
- Over-write all the num_classes from default value to your classes number.
```
model = dict(
    roi_head=dict(
        bbox_head=[
            dict(
                type='Shared2FCBBoxHead',
                num_classes=6),
            dict(
                type='Shared2FCBBoxHead',
                num_classes=6),
            dict(
                type='Shared2FCBBoxHead',
                num_classes=6],
    mask_head=dict(num_classes=6)))
```

**Comment by 666tugege** [comment]
### 4. For yolact network, the classification loss of negative samples is expanded to three times of the original.

- According to the formula of focal loss, the classification loss of achieving negative samples is expanded to three times of the original one, then the parameters alpha=0.25 and loss_weight=1.0 need to be expanded by three times.
```
loss_cls=dict(
    type='FocalLoss',
    use_sigmoid=True,
    gamma=2.0,
    alpha=0.75,
    loss_weight=3.0)
```
